### PR TITLE
Use testcontainers - allow placeholders in config

### PR DESCRIPTION
**Pull request by simlmx**
## Description

* Use the `testcontainers` python library to deal with the test
  database. This somewhat simplifies the tests (especially the makefile).
* Allow ${VARIABLE} placeholers in the config.
* Allow specifying those variables in environment variables and/or in a
  .env file.
* Add a (work in progress) config for our "dev" production environment

## How Has This Been Tested?

* Ran tests
* Added tests
* Ran the `app.py` script on the DEV environment with the provided config

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

## TODO
- [ ] Merge #15 
- [ ] Make sure the way we deal with placeholders in the config and environment variable is consistent the OCF workflow
- [ ] Review

### CODE REVIEW

File: configurations/aws_conf.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+run_model_func: pv_site_production.models.psp.psp_model.get_model
+model_path: dev_model.pkl
+
+nwp:
+  cls: pv_site_production.data.nwp_data_sources.NwpDataSource
+  kwargs:
+    path: s3://nowcasting-nwp-development/data/latest.netcdf
+    storage_kwargs:
+      key: ${OCF_AWS_KEY}
```
Comment (peterdudfield) [inline configurations/aws_conf.yaml:9]: we should need these keys for live. It just has the AWS keys using the IAM role. Locally i use `aws configure` to add my keys. 

Leave it in for this PR, but maybe remove it else where

File: tests/conftest.py [hunk 2]
```diff
@@ -15,27 +17,29 @@
...
+from testcontainers.postgres import PostgresContainer
 
 
 @pytest.fixture(scope="session", autouse=True)
 def db_connection():
     """Create data connection"""
 
-    # This points to our test docker DB defined in the Makefile.
-    url = "postgresql+psycopg2://postgres-test:postgres-test@localhost:5460/psp-test"
+    with PostgresContainer("postgres:14.5") as postgres:
```
Comment (peterdudfield) [inline tests/conftest.py:27]: nice

**Review by peterdudfield — APPROVED** [review]

File: configurations/aws_conf.yaml (hunk 1 above)
Comment (simlmx) [reply configurations/aws_conf.yaml:9]: Yeah that works with me.

File: pv_site_production/app.py [hunk 3]
```diff
@@ -47,8 +49,13 @@ def main(
...
     """Main function"""
-    with open(config_path) as f:
-        config = yaml.safe_load(f)
+
+    _log.debug("Load the configuration file")
+    # Typically the configuration will contain many placeholders pointing to environment variables.
+    # We allow specifying them in a .env file. See the .env.dist for a list of expected variables.
+    # Environment variables still have precedence.
+    dotenv_variables = dotenv.dotenv_values()
+    config = load_config(config_path, {**dotenv_variables, **os.environ})
```
Comment (simlmx) [inline pv_site_production/app.py:58]:
```suggestion
    config = load_config(config_path, dotenv_variables | os.environ)
```

File: configurations/aws_conf.yaml (hunk 1 above)
Comment (simlmx) [reply configurations/aws_conf.yaml:9]: Removing it in this PR!

**Comment by simlmx** [comment]
Merged in 8e8cce95a6b1861797b8524aad728b1bde93e2e3

**Pull request closed without merging**